src/main/groovy/com/lesfurets/jenkins/unit/DeclarativePipelineHarness.groovy

```diff
@@ -0,0 +1,164 @@
+package com.lesfurets.jenkins.unit
+
+class DeclarativePipelineHarness {
+    BasePipelineTest base
+
+    Map<String, CallStack> callStacks = [:]
+    Map<String, IdentifiedCallStack<?>> identifiedCallStacks = [:]
+
+    DeclarativePipelineHarness(base) {
+        this.base = base
+        registerCallStack('pipeline')
+        registerCallStack('options')
+        registerCallStack('parameters')
+        registerCallStack('environment')
+        registerCallStack('triggers')
+        registerCallStack('tools')
+        registerCallStack('input')
+
+        registerCallStack('agent')
+        registerCallStack('docker')
+        registerCallStack('dockerfile')
+        registerCallStack('node')
+
+        registerCallStack('stages')
+        registerCallStack('parallel')
+        registerCallStack('script')
+        registerCallStack('steps')
+
+        registerCallStack('when')
+        registerCallStack('allOf')
+        registerCallStack('anyOf')
+        registerCallStack('expression')
+
+        //Register post methods
+        registerCallStack('post')
+        registerCallStack('always')
+        registerCallStack('cleanup')
+        registerCallStack('success')
+        registerCallStack('failure')
+        registerCallStack('regression')
+        registerCallStack('changed')
+        registerCallStack('fixed')
+        registerCallStack('aborted')
+        registerCallStack('unstable')
+        registerCallStack('unsuccessful')
+
+        registerIdentifiedCallStack('stage', String.class)
+        registerIdentifiedCallStack('node', String.class)
+        registerIdentifiedCallStack('withCredentials', Object.class)
+        registerIdentifiedCallStack('withEnv', List.class)
+        registerIdentifiedCallStack('dir', String.class)
+    }
+
+    void runCall(String name, Integer idx = 0, Boolean doClear = true) {
+        CallStack callStack = callStacks[name]
+        if(callStack== null) {
+            throw new RuntimeException("No such call stack ${name}")
+        }
+        callStack.run(idx, doClear)
+    }
+
+    public <T> void runIdentified(String name, T value, Boolean doClear = true) {
```

Comment:
This method name sucks; but I couldn't come up with a bette one.

Maybe `runIdentifiedCall` or `runCallById` (which would also play nicely with `runCallByIdx`).

Reply:
renamed to `runClosureStep`